test/sql/test_agg_function/T/test_approx_top_k

```diff
@@ -0,0 +1,506 @@
+SELECT c_partition, APPROX_TOP_K(c_decimal, 1) OVER(PARTITION BY c_partition) FROM t_partition ORDER BY c_partition;
+
+-- name: test_approx_top_k_for_boolean
+CREATE TABLE `t_bool` (
+  `c_id` INT(11) NOT NULL,
+  `c_partition` INT(11) NOT NULL,
+  `c_bool` BOOLEAN NULL
+) ENGINE=OLAP
+DUPLICATE KEY(`c_id`)
+DISTRIBUTED BY HASH(`c_id`) BUCKETS 10
+PROPERTIES (
+ "replication_num" = "1"
+);
+
+INSERT INTO `t_bool` (
+  `c_id`, `c_partition`, `c_bool`) 
+VALUES 
+  -- Partition 1
+  (1, 1, true),
+  (2, 1, true),
+  (3, 1, true),
+  (4, 1, true),
+  (5, 1, NULL),
+  (6, 1, true),
+  (7, 1, false),
+  (8, 1, false),
+  (9, 1, NULL),
+  (10, 1, false),
+  (11, 1, false),
+
+  -- Partition 2
+  (21, 2, true),
+  (22, 2, true),
+  (23, 2, true),
+  (24, 2, true),
+  (25, 2, true),
+  (26, 2, false),
+  (27, 2, false),
+  (28, 2, false),
+  (29, 2, false),
+
+  -- Partition 3
+  (31, 2, NULL),
+  (32, 2, NULL),
+  (33, 2, NULL),
+  (34, 2, NULL),
+  (35, 2, NULL);
+
+-- Aggregation
+SELECT /*+ SET_VAR(streaming_preaggregation_mode='auto')*/ APPROX_TOP_K(c_bool) FROM t_bool;
+SELECT /*+ SET_VAR(streaming_preaggregation_mode='force_preaggregation')*/ APPROX_TOP_K(c_bool, 1) FROM t_bool;
+SELECT /*+ SET_VAR(streaming_preaggregation_mode='force_streaming')*/ APPROX_TOP_K(c_bool, 3, 100) FROM t_bool;
+SELECT /*+ SET_VAR(streaming_preaggregation_mode='auto')*/ c_partition, APPROX_TOP_K(c_bool) FROM t_bool GROUP BY c_partition ORDER BY c_partition;
+SELECT /*+ SET_VAR(streaming_preaggregation_mode='force_preaggregation')*/ c_partition, APPROX_TOP_K(c_bool, 1) FROM t_bool GROUP BY c_partition ORDER BY c_partition;
+SELECT /*+ SET_VAR(streaming_preaggregation_mode='force_streaming')*/ c_partition, APPROX_TOP_K(c_bool, 3, 100) FROM t_bool GROUP BY c_partition ORDER BY c_partition;
+
+-- Window Function
+SELECT c_id, APPROX_TOP_K(c_bool) OVER() FROM t_bool ORDER BY c_id;
+SELECT c_id, APPROX_TOP_K(c_bool, 10) OVER() FROM t_bool ORDER BY c_id;
+SELECT c_id, APPROX_TOP_K(c_bool, 1, 10) OVER() FROM t_bool ORDER BY c_id;
+SELECT c_partition, APPROX_TOP_K(c_bool) OVER(PARTITION BY c_partition) FROM t_bool ORDER BY c_partition;
+SELECT c_partition, APPROX_TOP_K(c_bool, 10) OVER(PARTITION BY c_partition) FROM t_bool ORDER BY c_partition;
+SELECT c_partition, APPROX_TOP_K(c_bool, 1, 10) OVER(PARTITION BY c_partition) FROM t_bool ORDER BY c_partition;
+
+-- name: test_approx_top_k_tpch
+function: prepare_data("tpch", "${db[0]}")
+SELECT APPROX_TOP_K(L_LINENUMBER) FROM lineitem;
+SELECT APPROX_TOP_K(L_LINENUMBER, 1) FROM lineitem;
+SELECT APPROX_TOP_K(L_LINENUMBER, 10) FROM lineitem;
+SELECT L_RETURNFLAG, APPROX_TOP_K(L_LINENUMBER) FROM lineitem GROUP BY L_RETURNFLAG ORDER BY L_RETURNFLAG;
+SELECT L_RETURNFLAG, APPROX_TOP_K(L_LINENUMBER, 2) FROM lineitem GROUP BY L_RETURNFLAG ORDER BY L_RETURNFLAG;
+SELECT L_RETURNFLAG, APPROX_TOP_K(L_LINENUMBER, 20) FROM lineitem GROUP BY L_RETURNFLAG ORDER BY L_RETURNFLAG;
```

Comment:
are there special inputs? const inputs, only null columns?

Reply:
yes, the above cases.